hledger-lib/Hledger/Data/Types.hs

```diff
@@ -230,7 +230,8 @@ data Amount = Amount {
       aprice      :: !(Maybe AmountPrice)  -- ^ the (fixed, transaction-specific) price for this amount, if any
     } deriving (Eq,Ord,Generic,Show)
 
-newtype MixedAmount = Mixed [Amount] deriving (Eq,Ord,Generic,Show)
+newtype MixedAmount = Mixed (M.Map MixedAmountKey Amount) deriving (Eq,Ord,Generic,Show)
```

Comment:
We usually seem to end up replacing Map with HashMap - any thoughts ?

Reply:
Could do. We would lose the ordering information, which I believe would make the commodity when you combine multiple zero amounts unpredictable. Perhaps not the biggest issue, especially since this step is done only when calling `amounts` after this PR, so we could do the more expensive operation of sorting at that stage.

Reply:
One issue is that HashMap does not have `partition` or `partitionWithKey` defined, both of which we use here. It seems this is just a matter of them not having been implemented yet, rather than any fundamental difficulties.